src/hooks/use-schema.js

```diff
@@ -0,0 +1,14 @@
+import { useContext, useEffect, useState } from 'react';
+import {LocationContext} from "../contexts/location";
+
+const useSchema = (forPath = []) => {
+  const { responseDocument } = useContext(LocationContext);
+  const [schema, setSchema] = useState(null);
+  useEffect(() => {
+    if (responseDocument)
+      responseDocument.getSchema(forPath).then(setSchema);
+  }, [responseDocument]);
+  return schema;
+};
+
+export default useSchema;
```

Comment:
nit: should we name files with a hyphen or camelCase ? we have both now, but the existing hooks were camelCase.